kong/router.lua

```diff
@@ -181,9 +185,9 @@ sort(SORTED_MATCH_RULES, function(a, b)
 end)
 
 local MATCH_SUBRULES = {
-  HAS_REGEX_URI    = 0x01,
-  PLAIN_HOSTS_ONLY = 0x02,
-  HAS_HOST_PORT    = 0x04,
+  HAS_REGEX_URI          = 0x01,
+  PLAIN_HOSTS_ONLY       = 0x02,
+  HAS_WILDCARD_HOST_PORT = 0x04,
```

Comment:
Renamed this submatching rule to be more precise: we only case about it when dealing with regex hosts.